examples/siro_sandbox/app_states/app_state_tutorial.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Meta Platforms, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import magnum as mn
+from app_states.app_state_abc import AppState
+from camera_helper import CameraHelper
+from hablab_utils import get_agent_art_obj_transform
+from hitl_tutorial import Tutorial, generate_tutorial
+
+from habitat.gui.gui_input import GuiInput
+from habitat.gui.text_drawer import TextOnScreenAlignment
+
+
+class AppStateTutorial(AppState):
+    def __init__(
+        self,
+        sandbox_service,
+        gui_agent_ctrl,
+    ):
+        self._sandbox_service = sandbox_service
+        self._gui_agent_ctrl = gui_agent_ctrl
+        self._cam_transform = None
+        self._camera_helper = CameraHelper(
+            self._sandbox_service.args, self._sandbox_service.gui_input
+        )
+
+    def get_sim(self):
+        return self._sandbox_service.sim
+
+    def get_gui_controlled_agent_index(self):
+        return self._gui_agent_ctrl._agent_idx
+
+    def _get_camera_lookat_pos(self):
+        agent_root = get_agent_art_obj_transform(
+            self.get_sim(), self.get_gui_controlled_agent_index()
+        )
+        lookat = agent_root.translation + mn.Vector3(0, 1, 0)
+        return lookat
+
+    def on_environment_reset(self, episode_recorder_dict):
+        base_pos = self._get_camera_lookat_pos()
```

Comment:
>I had another idea here that AppStateTutorial and AppStateRearrange could directly share a CameraHelper instance. That may be a more elegant way to sync the "final camera".

Instead of sharing CameraHelper, I just create CameraHelper instance call `self._camera_helper.get_cam_transform()` and `self._camera_helper_get_eye_and_lookat()` to get initial cam_transform and (eye_pos, lookat_pos). The same calculation will be done at AppStateRearrange.on_environment_reset().

Reply:
But how does this sync the camera from the rearrange app state? Are they just coincidentally the same right now? What if we change the camera logic in AppStateRearrange? What if we want to re-use this tutorial with another app state?